#! /bin/sh -e

# DP: hotshot: Check for the availability of the profile and pstats modules.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Lib/hotshot/stats.py~	2007-02-16 17:57:29.000000000 +0100
+++ Lib/hotshot/stats.py	2007-03-01 22:07:02.000000000 +0100
@@ -1,7 +1,10 @@
 """Statistics analyzer for HotShot."""
 
-import profile
-import pstats
+try:
+    import profile
+    import pstats
+except ImportError, e:
+    raise ImportError, str(e) + '; please install the python-profiler package'
 
 import hotshot.log
 
